docs/Ingress.md

```diff
@@ -1,59 +1,876 @@
+| [Let's Encrypt](#letsencrypt)                         |
+
+
+
+# Configuring Ingress <a name="introduction"></a>
 
 Using `port-forward` is great for testing, but you will ultimately want to
 make it easier to access your Splunk cluster outside of Kubernetes. A common 
-approach is using
-[Kubernetes Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/).
+approach is using [Kubernetes Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/).
+
+There are many Ingress Controllers available, with each having their own pros and cons. There are just as many ways to configure each of them, which also depends upon your specific infrastructure and organizational policies. Splunk Operator will automatically create and manage Kubernetes Services for all of the relevant components, and we expect these will provide for easy integration with most Ingress Controllers and configurations.
+
+```
+$ kubectl get services -o name
+service/splunk-cluster-cluster-master-service
+service/splunk-cluster-deployer-service
+service/splunk-cluster-indexer-headless
+service/splunk-cluster-indexer-service
+service/splunk-cluster-license-master-service
+service/splunk-cluster-search-head-headless
+service/splunk-cluster-search-head-service
+service/splunk-standalone-standalone-headless
+service/splunk-standalone-standalone-service
+```
+
+We provide some examples below for configuring a few of the most popular Ingress controllers: [Istio](https://istio.io/) , [Nginx-inc](https://docs.nginx.com/nginx-ingress-controller/overview/) and [Ingress Nginx](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration). We hope these will serve as a useful starting point to configuring ingress in your environment.
+
+Before deploying an example, you will need to review the yaml and replace “example.com” with the domain name you would like to use and replace “example” in the service names with the name of your custom resource object. You will also need to point your DNS for all the desired hostnames to the IP addresses of your ingress load balancer.
+
+
+#### Important Notes on using Splunk on Kubernetes 
+
+#### Splunk default network ports
+When creating a new Splunk instance on Kubernetes, the default network ports will be used for internal communication such as internal logs, replication, and others. Any change in how these ports are configured needs to be consistent across the cluster. 
+
+For Ingress we recommend using separate ports for encrypted and non-encrypted traffic. In this documentation we will use port 9998 for encrypted data coming from outside the cluster, while keeping the default 9997 for non-encrypted intra-cluster communication.
+
+#### Indexer Discovery not supported
+Indexer Discovery is not supported on a Kubernetes cluster. Instead, the Ingress controllers will be responsible to connect forwarders to peer nodes in Indexer clusters.
+
```

Comment:
Unless I missed it we haven't explicitly stated anywhere that any load balancer that is used must resolve to multiple IP addresses so that forwarder's autoLB feature works. it seems like this section of "important notes" would be a good place to put it.

Reply:
Done